Assets/Scripts/TextDecoder/ActionDecoder.cs

```diff
@@ -0,0 +1,416 @@
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Reflection;
+using System.Text.RegularExpressions;
+using UnityEngine;
+using UnityEngine.SceneManagement;
+
+public class ActionDecoder
+{
+    public event Action OnActionDone;
+    public IActorController ActorController { get; set; }
+    public ISceneController SceneController { get; set; }
+    public IAudioController AudioController { get; set; }
+    public IEvidenceController EvidenceController { get; set; }
+    public IAppearingDialogueController AppearingDialogueController { get; set; }
+
+    /// <summary>
+    ///     Parse action lines inside from inside .ink files
+    /// </summary>
+    /// <param name="actionLine">Line of a .ink file that starts with &amp; (and thereby is not a "spoken dialogue" line)</param>
+    /// <remarks>
+    ///     Writers are able to call methods inside .ink files. This is done by using the following syntax:
+    ///     <code>
+    ///     &amp;{methodName}:{parameter1},{parameter2},...
+    ///     </code>
+    ///     This method is responsible for:
+    ///         1. Finding a method inside this class, matching `methodName`
+    ///         2. Verifying the amount of parameters matches the amount of parameters needed in the method
+    ///         3. Attempting to parse each parameter into the correct type using <see cref="Parser&lt;T&gt;"/> of the type
+    ///         4. Invoking the method with the parsed parameters
+    /// </remarks>
+    public void OnNewActionLine(string actionLine)
+    {
+        actionLine = actionLine.Trim();
+        const char actionSideSeparator = ':';
+        const char actionParameterSeparator = ',';
+
+        string[] actionNameAndParameters = actionLine.Substring(1, actionLine.Length - 1).Trim().Split(actionSideSeparator);
+
+        if (actionNameAndParameters.Length > 2)
+        {
+            throw new TextDecoder.Parser.ScriptParsingException($"More than one '{actionSideSeparator}' detected in line '{actionLine}'");
+        }
+
+        string action = actionNameAndParameters[0];
+        string[] parameters = (actionNameAndParameters.Length == 2) ? actionNameAndParameters[1].Split(actionParameterSeparator) : Array.Empty<string>();
```

Comment:
```suggestion
        string[] actionNameAndParameters = actionLine.Substring(1, actionLine.Length - 1).Trim().Split(':');

        if (actionNameAndParameters.Length > 2)
        {
            throw new TextDecoder.Parser.ScriptParsingException($"More than one ':' detected in line '{actionLine}'");
        }

        string action = actionNameAndParameters[0];
        string[] parameters = (actionNameAndParameters.Length == 2) ? actionNameAndParameters[1].Split(',') : Array.Empty<string>();
```
Since these variables are only used a couple of times, would it be better to use the strings directly?

Reply:
This file is not supposed to be here. Deleted it.